rpc/wrtc_signaling_server.go

```diff
@@ -470,9 +467,21 @@ func (srv *WebRTCSignalingServer) Answer(server webrtcpb.SignalingService_Answer
 		}
 	}
 
+	srv.callMu.RLock()
+	// Atomically check if the cancelCtx was canceled, and if not, add to the `callWorkers`.
+	if err := srv.cancelCtx.Err(); err != nil {
+		srv.callMu.RUnlock()
+		return srv.cancelCtx.Err()
+	}
+	srv.callWorkers.Add(1)
+	srv.callMu.RUnlock()
+
 	callerErrCh := make(chan error, 1)
 	utils.PanicCapturingGo(func() {
-		defer close(callerErrCh)
+		defer func() {
+			close(callerErrCh)
```

Comment:
This one was the bug in the test

Reply:
Sorry which test? The data race you saw yesterday? We weren't keeping track of the `callerLoop` as a worker and it was trying to log after signal server close?

Reply:
Correct, https://viam.atlassian.net/browse/RSDK-8529 was a "data race" because of logging after the test ended. And that was due this goroutine going unaccounted for.